core/unit_test/tools/TestLogicalSpaces.hpp

```diff
@@ -131,7 +131,7 @@ void test_malloc_free() {
   auto* temp =
       Kokkos::kokkos_malloc<fake_memory_space>("does_malloc_work", 1000);
   expect_deallocation_event("does_malloc_work", "TestSpace", "Error in free");
-  Kokkos::kokkos_free(temp);
+  Kokkos::kokkos_free<fake_memory_space>(temp);
```

Comment:
Why do you have this change.  This was fixed in #5567

Reply:
I am adding a commit merging with `upstream/develop` now.